File: ERP_SERVER/ERP_Anass_backend/Models/JournalEntry.cs
using System.ComponentModel.DataAnnotations;

namespace ERP_Anass_backend.Models
{
    public enum JournalEntryStatus
    {
        Draft,
        Posted
    }

    public class JournalEntry : BaseEntity
    {
        [Key]
        public int idJournalEntry { get; set; }
        public DateTime EntryDate { get; set; } = DateTime.UtcNow;
        public string Description { get; set; }
        public float TotalDebit { get; set; }
        public float TotalCredit { get; set; }
        public JournalEntryStatus Status { get; set; } = JournalEntryStatus.Draft;

        public ICollection<JournalDetails> JournalDetails { get; set; }
    }
}

File: ERP_Anass_app/src/app/models/Finance/JournalEntry.ts

import { JournalDetails } from './JournalDetails';

export enum JournalEntryStatus {
    Draft = 0,
    Posted = 1
}

export interface JournalEntry {
    idJournalEntry?: number;
    entryDate?: Date;
    description?: string;
    totalDebit?: number;
    totalCredit?: number;
    status?: JournalEntryStatus;
    journalDetails?: JournalDetails[];
}
